scripts/utils/internals.js

```diff
@@ -136,6 +147,90 @@ function signTransaction(lw, signers, transactionHash) {
   return signatureBytes
 }
 
+const encodeMultiSend = async function(multiSend, txs, web3 = web3) {
+  return await multiSend.contract.methods
+    .multiSend(
+      `0x${txs
+        .map(tx =>
+          [
+            web3.eth.abi.encodeParameter("uint8", tx.operation).slice(-2),
+            web3.eth.abi.encodeParameter("address", tx.to).slice(-40),
+            web3.eth.abi.encodeParameter("uint256", tx.value).slice(-64),
+            web3.eth.abi.encodeParameter("uint256", web3.utils.hexToBytes(tx.data).length).slice(-64),
+            tx.data.replace(/^0x/, ""),
+          ].join("")
+        )
+        .join("")}`
+    )
+    .encodeABI()
+}
+
+/**
+ * Given a collection of transactions, creates a single transaction that bundles all of them
+ * @param {Transaction[]} transactions List of {@link Transaction} that are to be bundled together
+ * @return {Transaction} Multisend transaction bundling all input transactions
+ */
+const getBundledTransaction = async function(transactions, web3 = web3, artifacts = artifacts) {
+  const MultiSend = artifacts.require("MultiSend")
+  const multiSend = await MultiSend.deployed()
+  const transactionData = await encodeMultiSend(multiSend, transactions, web3)
+  const bundledTransaction = {
+    operation: DELEGATECALL,
+    to: multiSend.address,
+    value: 0,
+    data: transactionData,
+  }
+  return bundledTransaction
+}
+
+const execTransactionData = async function(gnosisSafeMasterCopy, owner, transaction) {
+  const sigs =
+    "0x" +
+    "000000000000000000000000" +
+    owner.replace("0x", "") +
+    "0000000000000000000000000000000000000000000000000000000000000000" +
+    "01"
+  return await gnosisSafeMasterCopy.contract.methods
+    .execTransaction(
+      transaction.to,
+      transaction.value,
+      transaction.data,
+      transaction.operation,
+      0,
+      0,
+      0,
+      ADDRESS_0,
+      ADDRESS_0,
+      sigs
+    )
+    .encodeABI()
+}
+
+/**
+ * Creates a transaction that makes a master Safe execute a transaction on behalf of a (single-owner) owned trader using execTransaction
+ * @param {EthereumAddress} masterAddress Address of a controlled Safe
+ * @param {EthereumAddress} traderAddress Address of a Safe, owned only by master, target of execTransaction
+ * @param {Transaction} transaction The transaction to be executed by execTransaction
+ * @return {Transaction} Transaction calling execTransaction; should be executed by master
+ */
+const getExecTransactionTransaction = async function(masterAddress, traderAddress, transaction, web3, artifacts) {
```

Comment:
Here, most likely the replace function did not work as intended

Reply:
Oh it was intended.

I would call the upper function `getExecutionTransactionData' and this one here 'getExecuctionTransaction'

Reply:
I sticked to the rule that if a function creates a function that runs `operation`, then the function is called `getOperationTransaction`. The problem is that here the operation was called `execTransaction`!
Bit I very much dislike the name (and this naming convention) so I'll definitely change that.